provisioning/miniprov/hare_mp/main.py

```diff
@@ -288,60 +277,83 @@ def save(filename: str, contents: str) -> None:
+    parser.add_argument('--report-unavailable-features',
+                        help='Report unsupported features according to setup type',
+                        action='store_true')
+    parser.add_argument('--configure-logrotate',
+                        help='Configure logrotate for hare',
+                        action='store_true')
+
+    parser = subparser.add_parser('config',
+                                  help='Configure Hare')
+    parser.set_defaults(func=config)
+    parser.add_argument('--config',
+                        help='Conf Store URL with cluster info',
+                        required=True,
+                        nargs=1,
+                        type=str,
+                        action='store')
+    parser.add_argument('--filepath',
+                        help='Full path to the CDF file to generate at this stage.',
+                        nargs=1,
+                        default=['/var/lib/hare/cluster.yaml'],
+                        type=str,
+                        action='store')
+
+    parser = subparser.add_parser('init',
+                                  help='Perform component initialization')
+    parser.set_defaults(func=init)
+    parser.add_argument('--filepath',
+                   help='Full path to the CDF file.',
                    nargs=1,
-                   default='/var/lib/hare/cluster.yaml',
+                   default=['/var/lib/hare/cluster.yaml'],
                    type=str,
                    action='store')
-    p.add_argument('--config',
-                   help='Configure Hare',
+
+    parser = subparser.add_parser('test',
+                                  help='Testing cluster status')
+    parser.set_defaults(func=test)
+    parser.add_argument('--filepath',
+                   help='Full path to the CDF file.',
                    nargs=1,
+                   default=['/var/lib/hare/cluster.yaml'],
                    type=str,
-                   default='',
-                   dest='config_url',
                    action='store')
+
+    parser = subparser.add_parser('support_bundle',
+                                  help='Generating support bundle')
+    parser.set_defaults(func=generate_support_bundle)
+
+    parser = subparser.add_parser('reset',
+                                  help='Reset/cleanup step')
+    parser.set_defaults(func=cleanup)
+    parser = subparser.add_parser('cleanup',
+                                  help='Reset/cleanup step')
+    parser.set_defaults(func=cleanup)
+
     setup_logging()
 
     parsed = p.parse_args(sys.argv[1:])
-    if parsed.config_url:
-        # --config also depends on optional parameter --filename,
-        # that is why the approach with putting the whole confi
-        # logic into the ConfigAction class will not work.
-        url = parsed.config_url[0]
-        filename = parsed.filename[0] or '/var/lib/hare/cluster.yaml'
-        save(filename, generate_cdf(url))
+
+    if parsed.func:
```

Comment:
Is there a case where `parsed.func` is not set? 
If yes, then what is the expected behavior? is it okay to exit silently? I think this means that the subcommand is not supported, shouldn't we throw and exception or log and error message in this case?

If no, then we need not have an `if` loop.

Reply:
Done